RFM9x.nut

```diff
@@ -0,0 +1,399 @@
+    static CAD_DONE = 2;
+    static FHSS_CHANGE_CHANNEL = 1;
+    static CAD_DETECTED = 0;
+    
+    // IRQ
+    static REG_IRQ_FLAGS = 0x12;
+    
+    static REG_RX_N_BYTES = 0x13; // number of bytes in most recent payload
+    
+    static REG_RX_HEADER_CNT_VALUE_MSB = 0x14;
+    static REG_RX_HEADER_CNT_VALUE_LSB = 0x15;
+    static REG_RX_PACKET_CNT_VALUE_MSB = 0x16;
+    static REG_RX_PACKET_CNT_VALUE_LSB = 0x17;
+    
+    static REG_MODEL_STAT = 0x18;
+    static REG_PKT_SNR_VALUE = 0x19;
+    static REG_PKT_RSSI_VALUE = 0x1A;
+    static REG_RSSI_VALUE = 0x1B;
+    static REG_HOP_CHANNEL = 0x1C;
+    static REG_MODEM_CONFIG1 = 0x1D;
+    static REG_MODEM_CONFIG2 = 0x1E;
+    static REG_MODEM_CONFIG3 = 0x26;
+    static REG_SYMB_TIMEOUT_LSB = 0x1F;
+    
+    static REG_PREAMBLE_MSB = 0x20;
+    static REG_PREAMBLE_LSB = 0x21;
+    static REG_PAYLOAD_LENGTH = 0x22;
+    static REG_MAX_PAYLOAD_LENGTH = 0x23;
+    static REG_HOP_PERIOD = 0x24;
+    static REG_FIFO_RX_BYTE_ADDR = 0x25;
+    
+    
+    // -------------------- Shared Registers ---------------------- //
+
+    static REG_DIO_MAPPING1 = 0x40;
+    static REG_DIO_MAPPING2 = 0x41;
+    static REG_VERSION = 0x42;
+    static REG_TCXO = 0x4B;
+    
+    // -------------------- Some Tables --------------------------- //
+    
+    static BWTABLE = {
+        "7.8" : 0x00,
+        "10.4" : 0x01,
+        "15.6" : 0x02,
+        "20.8" : 0x03,
+        "31.25": 0x04,
+        "41.7": 0x05,
+        "62.5": 0x06,
+        "125": 0x07,
+        "250" : 0x08,
+        "500": 0x09
+    };
+    
+    static CRTABLE = {
+        "4/5": 0x01,
+        "4/6": 0x02,
+        "4/7": 0x03,
+        "4/8": 0x04
+    };
+    
+    static SLEEP = 0x00;
+    static STANDBY = 0x01;
+    static FSTX = 0x02;
+    static TX = 0x03;
+    static FSRX = 0x04;
+    static RXCONTINUOUS = 0x05;
+    static RXSINGLE = 0x06;
+    static CAD = 0x07;
+    
+    static _txHeaderTo = 0xFF;
+    static _txHeaderFrom = 0x00;
+    static _txHeaderId = 0;
+    static _thisAddress = 0x66;
+    static _txHeaderFlags = 0;
+    
+    constructor(spi, irq, cs=null) {
+        // Assume spi is already initialized, assume cs is already initialized
+        this._spi = spi;
+        this._irqPin = irq;
+        this._cs = cs;
+    }
+    
+    function init() {
+        setMode(SLEEP); // need to do this to go into LoRa mode
+        
+        // Defaults
+        setFrequency(915000000);
+        setBandwidth("125");
+        setSpreadingFactor(7);
+        setCodingRate("4/5");
+        setImplicitHeaderMode(false);
+        setPreambleLength(8);
+        setPayloadLength(10);
+
+        configIRQ();
+        
+        _setInLoRaMode();
+
+    }
+
+    function configIRQ() {
+        if(_irqPin) {
+            _irqPin.configure(DIGITAL_IN_PULLDOWN, _isr.bindenv(this));
+        }
+        else {
+            throw "interrupt pin passed to constructor is invalid";
+        }
+    }
+
+    function _isr() {
```

Comment:
Please avoid using 3 letter abbreviation names for constants, members, functions. This makes the code unmaintainable.